bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuntimeCondition.java

```diff
@@ -26,7 +27,8 @@
     private Set<Connection> connections;
 
     public RuntimeCondition(Condition condition) {
-        super(condition.getId(), condition.getTypeUID(), condition.getConfiguration(), condition.getInputs());
+        super(condition.getId(), condition.getTypeUID(),
+                new Configuration(condition.getConfiguration().getProperties()), condition.getInputs());
```

Comment:
Yes you are right. We can use this method to update the configuration. The runtime object are not public and there for the method can't be call outside of rule engine.